### ISSUE TITLE
How to add additional OAuth Scopes / Permissions to existing account?

### ISSUE THREAD

**Issue by benoror** [nextauthjs/next-auth#2066]
## Question 💬

TL;DR; Need to grant additional OAuth scopes after the user have signed up/in, besides `email`

Use-case: After a user have signed up to our website, they can optionally opt-in to grant acces to Google Spreadsheets, so an additional scope such as `https://www.googleapis.com/auth/spreadsheets ...more` should be granted by opening a new Consent screen with extended permissions

I've seen in the docs callbacks can be used to things such as [Refresh Token Rotation](https://next-auth.js.org/tutorials/refresh-token-rotation) (which is something I would need as well), but I am still not sure how to use those for the incremental scope use-case, or how to simply trigger a new Consent screen with extended scopes.

- Google docs on incremental auth: https://developers.google.com/identity/sign-in/web/incremental-auth
- Related to: https://github.com/nextauthjs/next-auth/issues/1160, https://github.com/nextauthjs/next-auth/issues/1277, https://github.com/nextauthjs/next-auth/issues/408, https://github.com/nextauthjs/next-auth/discussions/1040

Ay hit would be greatly appreciated! Thanks 🙏

**Comment by balazsorban44** [comment]
So if I am not mistaken, the documentation you refer to make the user go through a sign-in process again, but this time with additional scopes in the URL. Access tokens once issued, cannot be modified. You will need a new one with new scopes in it.

This can be achieved with the `signIn` method, where the third argument is the parameters you want to pass. So you can add your new scopes there as additional request params.

https://datatracker.ietf.org/doc/html/rfc6749#section-3.3

https://next-auth.js.org/getting-started/client#additional-params
